packages/common/src/services/pagination.service.ts

```diff
@@ -184,23 +208,44 @@ export class PaginationService {
 
   goToFirstPage(event?: any, triggerChangeEvent = true): Promise<ServicePagination> {
     this._pageNumber = 1;
-    return triggerChangeEvent ? this.processOnPageChanged(this._pageNumber, event) : Promise.resolve(this.getFullPagination());
+    if (triggerChangeEvent) {
+      return this.cursorBased && this._pageInfo
+        ? this.processOnPageChanged(this._pageNumber, event, { newPage: this._pageNumber, pageSize: this._itemsPerPage, first: this._itemsPerPage })
+        : this.processOnPageChanged(this._pageNumber, event);
+    }
+    return Promise.resolve(this.getFullPagination());
   }
 
   goToLastPage(event?: any, triggerChangeEvent = true): Promise<ServicePagination> {
     this._pageNumber = this._pageCount || 1;
-    return triggerChangeEvent ? this.processOnPageChanged(this._pageNumber || 1, event) : Promise.resolve(this.getFullPagination());
+    if (triggerChangeEvent) {
+      return this.cursorBased && this._pageInfo
+        ? this.processOnPageChanged(this._pageNumber, event, { newPage: this._pageNumber, pageSize: this._itemsPerPage, last: this._itemsPerPage })
+        : this.processOnPageChanged(this._pageNumber, event);
+    }
+    return Promise.resolve(this.getFullPagination());
   }
 
   goToNextPage(event?: any, triggerChangeEvent = true): Promise<boolean | ServicePagination> {
     if (this._pageNumber < this._pageCount) {
       this._pageNumber++;
-      return triggerChangeEvent ? this.processOnPageChanged(this._pageNumber, event) : Promise.resolve(this.getFullPagination());
+      if (triggerChangeEvent) {
+        return this.cursorBased && this._pageInfo
+          ? this.processOnPageChanged(this._pageNumber, event, { newPage: this._pageNumber, pageSize: this._itemsPerPage, first: this._itemsPerPage, after: this._pageInfo.endCursor })
+          : this.processOnPageChanged(this._pageNumber, event);
+      } else {
+        return Promise.resolve(this.getFullPagination());
+      }
     }
     return Promise.resolve(false);
   }
 
   goToPageNumber(pageNumber: number, event?: any, triggerChangeEvent = true): Promise<boolean | ServicePagination> {
+    if (this.cursorBased) {
+      console.assert(true, 'Cursor based navigation cannot navigate to arbitrary page');
```

Comment:
oh that's interesting, I've never used this type of console, so it looks like a `console.error` when the assertion is failing. That could be interesting to use in the future. I guess it's also easier for unit tests since you don't need an `if` condition wrapper and so less lines of code as untested haha